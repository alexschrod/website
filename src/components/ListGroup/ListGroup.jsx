import React from "react"
import "./ListGroup.css"

const ListGroup = ({ values, columnSize, icon }) => {
  columnSize = columnSize || 4
  icon = icon || "hand-o-right"

  const listEntries = []
  values.forEach(element => {
    listEntries.push(
      <li key={element} className="list-group-item">
        <i className={"fa fa-" + icon}></i> {element}
      </li>
    )
  })

  return (
    <div className={"col-sm-" + columnSize}>
      <ul className="list-group">{listEntries}</ul>
    </div>
  )
}

export default ListGroup
