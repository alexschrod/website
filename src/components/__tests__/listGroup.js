import React from "react"
import renderer from "react-test-renderer"
import { shallow, mount, render } from "enzyme"

import ListGroup from "../ListGroup"

const values = ["One", "Two", "Three"]

describe("ListGroup", () => {
  it("should return null when no values are given", () => {
    const tree = renderer.create(<ListGroup values={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly", () => {
    const tree = renderer.create(<ListGroup values={values} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly with custom columnSize", () => {
    const tree = renderer
      .create(<ListGroup values={values} columnSize={6} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly with custom icon", () => {
    const tree = renderer
      .create(<ListGroup values={values} icon="icon" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
