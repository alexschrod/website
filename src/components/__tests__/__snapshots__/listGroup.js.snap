// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup renders correctly 1`] = `
<div
  className="col-sm-4"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      One
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      Two
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      Three
    </li>
  </ul>
</div>
`;

exports[`ListGroup renders correctly with custom columnSize 1`] = `
<div
  className="col-sm-6"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      One
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      Two
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-hand-o-right"
      />
       
      Three
    </li>
  </ul>
</div>
`;

exports[`ListGroup renders correctly with custom icon 1`] = `
<div
  className="col-sm-4"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-icon"
      />
       
      One
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-icon"
      />
       
      Two
    </li>
    <li
      className="list-group-item"
    >
      <i
        className="fa fa-icon"
      />
       
      Three
    </li>
  </ul>
</div>
`;

exports[`ListGroup should return null when no values are given 1`] = `null`;
